src/UNL/UCBCN/Manager/CreateEvent.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace UNL\UCBCN\Manager;
+
+use UNL\UCBCN\Calendar as CalendarModel;
+use UNL\UCBCN\Calendar\EventTypes;
+use UNL\UCBCN\Locations;
+use UNL\UCBCN\Event;
+use UNL\UCBCN\Event\EventType;
+use UNL\UCBCN\Event\Occurrence;
+use UNL\UCBCN\User;
+
+class CreateEvent 
+{
+	public $options = array();
+
+    public $calendar;
+
+    public function __construct($options = array()) 
+    {
+    	$this->options = $options + $this->options;
+        $this->calendar = CalendarModel::getByShortName($this->options['calendar_shortname']);
+
+        if (!empty($_POST)) {
+            error_log(print_r($_POST,1));
+            $this->saveEvent($_POST);
+            header('Location: /manager/' . $this->calendar->shortname . '/');
+        }
+
+        if ($this->calendar === FALSE) {
+            throw new \Exception("That calendar could not be found.", 500);
+        }
+    }
+
+    public function getEventTypes()
+    {
+        return new EventTypes(array());
+    }
+
+    public function getLocations()
+    {
+        $username = $_SESSION['__SIMPLECAS']['UID'];
+        return new Locations(array('user_id' => $username));
+    }
+
+    private function calculateDate($date, $hour, $minute, $am_or_pm) {
+        $date = strtotime($date);
+        # add hours correctly
+        # TODO: handle when this is not entered
+        $hours_to_add = (int)($hour) % 12;
+        if ($am_or_pm == 'pm') {
+            $hours_to_add += 12;
+        }
+        $date += $hours_to_add * 60 * 60 + (int)($minute) * 60;
+        return date('Y-m-d H:i:s', $date);
+    }
+
+    private function saveEvent($post_data) 
+    {
+        $username = $_SESSION['__SIMPLECAS']['UID'];
```

Comment:
I'm seeing this a lot.  I wonder if it might be better to add a method to the main controller or a session class named something like `::getCurrentUser()`, which would return the current user object or `false` if there isn't one.  Otherwise, we will have to update a lot of things if a different login method is ever used.